Remove version: Alibaba.AliIM version 9.12.12C

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #257406

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.10 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.10.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manual validation

- See https://github.com/microsoft/winget-pkgs/issues/257406#issuecomment-2888374545

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/257437)

Dragon1573 committed:
Bump version: Alibaba.AliIM version 9.12.12C -> 9.12.13C

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

ItzLevvie commented on manifests/a/Alibaba/AliIM/9.12.13C/Alibaba.AliIM.installer.yaml:
```diff
@@ -0,0 +1,27 @@
+- Architecture: x86
+  # There's something wrong with their server.
+  # There's NO installer URL for 9.12.13C, but the URL for 9.12.12C is actually 9.12.13C file.
+  InstallerUrl: https://download.alicdn.com/wangwang/AliIM_taobao_(9.12.12C).exe
```
```suggestion
  InstallerUrl: https://download.alicdn.com/wangwang/AliIM_taobao_(9.12.13C).exe
```

Dragon1573 committed:
Apply suggestions from code review

- Bump installer URL since they provide the correct URL.

Co-authored-by: Levvie - she/her <11600822+ItzLevvie@users.noreply.github.com>

ItzLevvie commented:
[Policy] Blocking-Issue

Dragon1573 commented:
> [!CAUTION]
> 
> Since there's a https://github.com/microsoft/winget-pkgs/labels/Blocking-Issue and got https://github.com/microsoft/winget-pkgs/labels/Binary-Validation-Error during tests, I'll try to only remove the outdated version without bumping to a newer one.

Dragon1573 committed:
Remove version: Alibaba.AliIM version 9.12.13C

- This version can't pass the pipeline.
- It will get `Binary-Validation-Error` from the pipeline because of false positive.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>
